components/proxy/src/main/kotlin/com/hotels/styx/services/HealthCheckMonitoringService.kt

```diff
@@ -85,12 +85,32 @@ internal class HealthCheckMonitoringService(
     }
 
     override fun stopService() = CompletableFuture.runAsync {
-        LOGGER.info("stopped")
+        LOGGER.info("stopped service for {}", application)
 
         objectStore.entrySet()
                 .filter(::containsRelevantStateTag)
-                .forEach { (name, _) ->
-                    markObject(objectStore, name, ObjectActive(0, false))
+                .forEach { (name, record) ->
+                    objectStore.get(name).ifPresent {
+                        try {
+                            objectStore.compute(name) { previous ->
+                                if (previous == null) throw ObjectDisappearedException()
+
+                                val newTags = record.tags
```

Comment:
Should this be `previous.tags` instead of `record.tags`?

Reply:
Either way. `previous` and `record` are the same object. But using `previous` feels more correct sine it may have been updated concurrently. Will fix this.